homeassistant/components/niu/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for Niu integration."""
+import logging
+
+from niu import NiuAPIException, NiuCloud, NiuNetException, NiuServerException
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+# Define data schema to be used and displayed by the config screen
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    try:
+        # Create new object with user data
+        hub = NiuCloud(
+            username=data[CONF_USERNAME],
+            password=data[CONF_PASSWORD],
+            lang="en-US",
+        )
+        # Check if this action have given us a correct auth token
+        if not await hub.connect():
+            _LOGGER.error("No authentication token found")
+            raise InvalidAuth
+
+    except NiuAPIException as ex:
+        _LOGGER.error("Error while authenticating with Niu API: %s", ex)
+        raise InvalidAuth from ex
+    except NiuServerException as ex:
+        _LOGGER.error("Error while making Niu API request: %s", ex)
+        raise InvalidAuth from ex
+    except NiuNetException as ex:
+        _LOGGER.error("Could not connect with Niu API: %s", ex)
+        raise CannotConnect from ex
+
+    # Return info that you want to store in the config entry.
+    return {"title": data[CONF_USERNAME]}
```

Comment:
We probably don't have to return anything here.

Reply:
Great suggestion! I've used the return for a token return but later I decided it wasn't a good idea to add the token to the config.